src/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -125,5 +114,64 @@ public static IEnumerable<string> GetAllProjectFileNames(string solutionFile)
                 throw new CommandLineException(message);
             }
         }
+
+        public static string GetMsbuildDirectory(string version)
+        {
+            if (String.IsNullOrEmpty(version))
+            {
+                // default to 4.0
+                version = "4.0";
+
+                // run msbuild to get the version
+                var processStartInfo = new ProcessStartInfo
+                {
+                    UseShellExecute = false,
+                    FileName = "msbuild.exe",
+                    Arguments = "/version",
+                    RedirectStandardOutput = true
+                };
+                try
+                {
+                    using (var process = Process.Start(processStartInfo))
+                    {
+                        process.WaitForExit(100);
+                        if (process.ExitCode == 0)
+                        {
+                            var output = process.StandardOutput.ReadToEnd();
+                            var lines = output.Split(
+                                new[] { Environment.NewLine },
+                                StringSplitOptions.RemoveEmptyEntries);
+                            version = lines[lines.Length - 1];
+                        }
+                    }
+                }
+                catch
+                {
+                }
+            }
+
+            var ver = version.Contains('.') ?
+                new Version(version) :
```

Comment:
This is the output of `msbuild /version` for msbuild 14:
```
Microsoft (R) Build Engine version 14.0.23107.0
Copyright (C) Microsoft Corporation. All rights reserved.

14.0.23107.0
```

And this is the output of msbuild 12:
```
Microsoft (R) Build Engine version 12.0.31101.0
[Microsoft .NET Framework, version 4.0.30319.42000]
Copyright (C) Microsoft Corporation. All rights reserved.

12.0.31101.0
```

Note that the code is not parsing the first line. Instead, it's returning the last line, which is the version.